File: src/main/java/co/q64/stars/block/BaseBlock.java
package co.q64.stars.block;

import lombok.Getter;
import net.minecraft.block.Block;
import net.minecraft.block.material.Material;

@Getter
public abstract class BaseBlock extends Block {
    private String id;

    public BaseBlock(String id) {
        this(id, Properties.create(Material.IRON));
    }

    public BaseBlock(String id, Material material) {
        this(id, Properties.create(Material.IRON));
    }

    public BaseBlock(String id, Properties settings) {
        super(settings);
        this.id = id;
        setRegistryName(id);
    }
}

File: src/main/kotlin/co/q64/stars/block/BaseBlock.kt
package co.q64.stars.block

import co.q64.stars.id
import net.minecraft.block.Block
import net.minecraft.block.SoundType
import net.minecraft.block.material.Material

val earth = Block.Properties.create(Material.EARTH).sound(SoundType.GROUND).hardnessAndResistance(0f, 0f)
val hard = Block.Properties.create(Material.IRON).sound(SoundType.METAL).hardnessAndResistance(-1f, 3600000f)

abstract class BaseBlock(blockId: String, properties: Properties = Properties.create(Material.IRON)) : Block(properties) {
    init {
        id = blockId
    }
}

interface HardBlock {}